spec/unit/shoulda/matchers/active_record/have_rich_text_matcher_spec.rb

```diff
@@ -0,0 +1,99 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveRichText, type: :model do
+  shared_examples 'undefined rich text' do |kind|
+    it 'raise a undefined method error when action text is not defined' do
+      create_table_posts
+  
+      invalid_record = -> { new_post(has_rich_text_association: true) }
+
+      message = /undefined method `has_rich_text' for Post/
+        
+      expect(&invalid_record).to raise_error(
+        NoMethodError,
+        message
+      )
+    end
+
+    include_examples "invalid attribute", kind
+  end
+
+  shared_examples "invalid attribute" do |kind|
+    it "does not matches when the provided attribute it's not a RichText class" do
+      create_table_posts(has_content_attribute: true)
+  
+      invalid_model = define_model_class(:Post)
+      invalid_record = invalid_model.new(content: "Content")
+      expected_message = "Expected Post to have :content as a " \
+        "ActionText::RichText but :content is a #{kind}"
+  
+      expect { have_rich_text(:content) }.
+        not_to match_against(invalid_record).
+        and_fail_with(expected_message)
+    end
+  end
+
+  def self.rich_text_is_defined?
+    defined?(ActionText::RichText)
+  end
+
+  if rich_text_is_defined?
+    it 'returns the message including the name of the provided attribute' do
+      matcher = have_rich_text(:content)
+      expect(matcher.description).
+        to eq('have :content as a ActionText::RichText')
+    end
+
+    context 'a has_rich_text association' do
+      include_examples "invalid attribute", 'String'
+
+      it 'matches when the subject configures has_rich_text for a custom attribute' do
+        create_table_posts
+    
+        valid_record = new_post(has_rich_text_association: true)
+    
+        expect(valid_record).to have_rich_text(:content)
+      end
+    
+      it 'does not matches when lacking has_rich_text attribute' do
+        create_table_posts
+    
+        invalid_record = new_post
+        expected_message = 'Expected Post to have :invalid_attribute as a ' \
+          'ActionText::RichText but :invalid_attribute does not exist'
+        
+        expect { have_rich_text(:invalid_attribute) }.
+          not_to match_against(invalid_record).
+          and_fail_with(expected_message)
+      end
+
+      it 'fails with the appropriate message when negated' do
```

Comment:
This should be covered already if you use the `match_against` matcher as I've mentioned on line 55.